api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +207,211 @@ message JwtHeader {
-  //   - headers:
-  //     - name: :method
-  //       value: OPTIONS
-  //   - headers:
-  //     - name: :path
-  //       regex_match: /.*
+  //    match:
+  //       prefix: "/"
   //
-  // Examples: bypass /healthz check
+  envoy.api.v2.route.RouteMatch match = 1 [(validate.rules).message.required = true];
+
+  // Specify a Jwt Requirement.
+  JwtRequirement requires = 2;
+}
+
+// This is the Envoy HTTP filter config for JWT authentication.
+//
+// For example:
+//
+// .. code-block:: yaml
+//
+//   providers:
+//      - id: provider1
+//        issuer: issuer1
+//        audiences:
+//        - audience1
+//        - audience2
+//        remote_jwks:
+//          http_uri:
+//            uri: https://example.com/.well-known/jwks.json
+//            cluster: example_jwks_cluster
+//      - id: provider2
+//        issuer: issuer2
+//        local_jwks:
+//          inline_string: jwks_string
+//
+//   rules:
+//      # Not jwt verification is required for /health path
+//      - match:
+//          prefix: "/health"
+//
+//      # Jwt verification for provider1 is required for path prefixed with "prefix"
+//      - match:
+//          prefix: "/prefix"
+//        requires:
+//          provider_id: "provider1"
+//
+//      # Jwt verification for either provider1 or provider2 is required for all other requests.
+//      - match:
+//          prefix: "/"
+//        requires:
+//          requires_any:
+//            requirements:
+//              - provider_id: "provider1"
+//              - provider_id: "provider2"
+//
+//// [#not-implemented-hide:]
+message JwtAuthentication {
+  // List of JWT providers. For example:
   //
   // .. code-block:: yaml
   //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: GET
-  //   - headers:
-  //     - name: :path
-  //       exact_match: /healthz
+  //   providers:
+  //      - id: provider1
+  //        issuer: issuer1
+  //        audiences:
+  //        - audience1
+  //        - audience2
+  //        remote_jwks:
+  //          http_uri:
+  //            uri: https://example.com/.well-known/jwks.json
+  //            cluster: example_jwks_cluster
+  //      - id: provider2
+  //        issuer: provider2
+  //        local_jwks:
+  //          inline_string: jwks_string
+  //
+  repeated JwtProvider providers = 1;
+
+  // If true, the request is allowed if JWT is mising or JWT verification fails.
+  // Default is false, a request without JWT or failed JWT verification is not allowed.
+  // A typical usage is: this filter is used to only verify JWT and pass the verified JWT
+  // payload to another filter, the other filter will make decision.
+  // This flag has impact on the process flow. For example, if flag is false, for multiple
+  // "one_of" requirements, if the first requirement passes, the rest requiements will not be
+  // verified. But if this flag is true, all requirements will be verified even all of them
+  // passed.
+  bool allow_missing_or_failed = 2;
```

Comment:
Can we nuke this flag by adding an empty requirement in `RequrirementRule` so `provider: issuser-A` OR  `empty` means same as this? Sounds like it gives more flexibility.

Reply:
Good idea.  will update it.